### PR TITLE
pachdev: Allow users to set a custom console version

### PR DESCRIPTION
**Pull request by jrockway**
This lets you pass `--console` to `push` to set a console version.  It's remembered for subsequent pushes.

### CODE REVIEW

File: src/internal/kindenv/kindenv.go [hunk 1]
```diff
@@ -43,9 +43,10 @@ const (
 	clusterHostnameKey     = "dev.pachyderm.io/hostname"
 
 	// Annotations that configure a namespace.
-	tlsKey            = "dev.pachyderm.io/tls"
-	exposedPortsKey   = "dev.pachyderm.io/exposed-ports"
-	portBindingPrefix = "dev.pachyderm.io/port."
+	tlsKey             = "dev.pachyderm.io/tls"
+	exposedPortsKey    = "dev.pachyderm.io/exposed-ports"
+	portBindingPrefix  = "dev.pachyderm.io/port."
+	consoleOverrideKey = "dev.pachyderm.io/console-version"
```
Comment (robert-uhl) [inline src/internal/kindenv/kindenv.go:49]: FWIW, I like using DNS subdomains for namespacing here.

File: src/internal/kindenv/kindenv.go [hunk 2]
```diff
@@ -227,6 +230,9 @@ func (c *Cluster) GetConfig(ctx context.Context, namespace string) (*ClusterConf
 	} else {
 		errors.JoinInto(&errs, errors.New("unable to determine whether or not to use TLS"))
 	}
+	if console, ok := ns.Annotations[consoleOverrideKey]; ok {
```
Comment (robert-uhl) [inline src/internal/kindenv/kindenv.go:233]: `ok && console != ""`?

**Review by robert-uhl — APPROVED** [review]
Couple of small questions/nits, but LGTM.

File: src/internal/kindenv/pachyderm.go [hunk 3]
```diff
@@ -104,6 +106,19 @@ func (c *Cluster) helmFlags(ctx context.Context, install *HelmConfig) ([]string,
 			"proxy.host=" + cfg.Hostname,
 		}, ","),
 	}
+	// Configure console.
+	if t := install.ConsoleTag; t != "" {
+		flags = append(flags, "--set", "console.image.tag="+t)
+		if err := c.editNamespace(ctx, install.Namespace, func(ns *corev1.Namespace) error {
+			ns.Annotations[consoleOverrideKey] = t
+			return nil
+		}); err != nil {
```
[the remark this answers is not in the document]
Comment (jrockway) [reply src/internal/kindenv/pachyderm.go:115]: I think it's pretty surprising if you set the console version, but it doesn't stick, then you push your changes and console rolls back to the helm-included version.

File: src/internal/kindenv/kindenv.go (hunk 2 above)
Comment (jrockway) [reply src/internal/kindenv/kindenv.go:233]: result.ConsoleTag is already "" (the zero value for a string), so we wouldn't change the outcome of the function if we added that check 😂

**Pull request merged**